src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.Extensions.Logging.Console.Internal
+{
+    /// <summary>
+    /// For non-Windows platform consoles which understand the ANSI escape code sequences to represent color
+    /// </summary>
+    public class AnsiLogConsole : IConsole
+    {
+        private readonly StringBuilder _outputBuilder;
+
+        public AnsiLogConsole()
+        {
+            _outputBuilder = new StringBuilder();
+        }
+
+        public void Write(string message, ConsoleColor? background, ConsoleColor? foreground)
+        {
+            // Order: backgroundcolor, foregroundcolor, Message, reset foregroundcolor, reset backgroundcolor
+            message = SetForegroundColor(message, foreground);
+            message = SetBackgroundColor(message, background);
+            _outputBuilder.Append(message);
+        }
+
+        public void WriteLine(string message, ConsoleColor? background, ConsoleColor? foreground)
+        {
+            Write(message, background, foreground);
+            _outputBuilder.AppendLine();
+        }
+
+        public void Flush()
+        {
+            System.Console.Write(_outputBuilder.ToString());
+            _outputBuilder.Clear();
+        }
+
+        private string SetForegroundColor(string message, ConsoleColor? foreground)
```

Comment:
You want to write these things to the builder instead of appending them as strings.